src/components/Camera/withFailureHandling.js

```diff
@@ -0,0 +1,51 @@
+// @flow
+import * as React from 'react'
+import { h, Component } from 'preact'
+import CameraError from '../CameraError'
+
+const generalError = { name: 'CAMERA_NOT_WORKING', type: 'error' }
+
+type State = {
+  hasError: boolean,
+}
+
+type Props = {
+  onError: ?string => void,
+}
+
+export default <WrappedProps: *>(
+  WrappedCamera: React.ComponentType<WrappedProps>
```

Comment:
hmm, so is this component mean to be used just with a Camera component? Because the `WrapperProps` type seems to be any, but the component is called `WrapperCamera`

Reply:
Yes, definitely it should always be a Camera component. Will update, thanks!

Reply:
Actually no 😔 , there are some shared props not defined in Camera but defined in CameraError, that will fail if we define a specific props type definition.

I guess it makes sense to not try to type check Camera here, since it's not really related to the inner workings of this HOC.